src/events/controller.py

```diff
@@ -0,0 +1,215 @@
+#!/usr/bin/env python3
```

Comment:
**praise** I really like that we have managers and handlers for the controller as well! Well done!